src/ConnectedMode/Persistence/ISolutionBindingRepository.cs

```diff
@@ -70,5 +86,42 @@ public BoundSonarQubeProject Read(string configFilePath)
 
             return bound;
         }
+
+        public bool Write(string configFilePath, BoundSonarQubeProject binding)
+        {
+            _ = binding ?? throw new ArgumentNullException(nameof(binding));
+
+            if (string.IsNullOrEmpty(configFilePath))
+            {
+                return false;
+            }
+
+            if (!solutionBindingFileLoader.Save(configFilePath, binding))
+            {
+                return false;
+            }
+
+            credentialsLoader.Save(binding.Credentials, binding.ServerUri);
+
+            return true;
+        }
+
+        public IEnumerable<BoundSonarQubeProject> List()
+        {
+            var result = new List<BoundSonarQubeProject>();
+
+            var bindingConfigPaths = unintrusiveBindingPathProvider.GetBindingPaths();
+
+            foreach (var bindingConfigPath in bindingConfigPaths)
+            {
+                var boundSonarQubeProject = Read(bindingConfigPath);
```

Comment:
Reusing the read method is good, but we might not need to load the credentials for each connection. You could create a private read method with a flag to load credentials